### ISSUE TITLE
FEATURE REQUEST:  Allow SNGREP To Be Used as a Live Monitoring Tool

### ISSUE THREAD

**Issue by kismetgerald** [irontec/sngrep#139]
I know SNGREP is designed to store packet capture to RAM by default, with an option to write to disk - that's all good.  But I have been using SNGREP as a live monitoring tool because it's just awesome that way (and looks cool too).  I also know you support capturing to a remote Homer instance (but this requires standing up a Homer server).

Anyway I find myself having to exit and relaunch SNGREP several times during the day when _**capture.limit**_ is reached.  Since my box is in a low volume enterprise environment, using SNGREP this way makes for a very useful monitoring and troubleshooting tool.  I have my system configured to filter only REGISTER and INVITE packets and this works flawlessly, but I did notice that ALL packets are still being captured, just not displayed.  This, therefore, contributes to me reaching the defined capture limit fairly quickly.

It would be nice if perhaps an option in Settings could be added such as:  _**capture.live**_ or _**capture.live_monitor**_.  What this would do is, when enabled, cause SNGREP to start discarding the oldest packets so that new packets are captured and not ignored as they currently are.

Another idea I had was perhaps to add support for capturing directly to a database (such as PostgreSQL or MySQL) which would then make it possible to use this awesome tool as a live monitoring tool (limited only by db performance).

**Comment by davemidd** [comment]
You mean like this https://github.com/irontec/sngrep/issues/94#issuecomment-222357078 :)

**Comment by kismetgerald** [comment]
@davemidd Yes, exactly.  Well without my second wish for db support, but yes.  So this is currently supported using _**sngrep -l 200 -R port 5060**_ ?

**Comment by davemidd** [comment]
@kismetgerald yep, that should do it.

**Comment by kismetgerald** [comment]
awesome!

**Comment by Kaian** [comment]
If you want to capture only dialogs starting with invite or register give it a try to command line match expression

`sngrep -l 200 -R "Cseq: [0-9]+ (INVITE|REGISTER)" port 5060`

**Comment by kismetgerald** [comment]
Hey @Kaian, welcome back.  hope vacation was awesome.
I tried the command above and sngrep responded with "_**Unknown option -R**_".  My system is update to date via apt-get dist-upgrade.  Anything special I need to do to get the changes you committed?

**Comment by davemidd** [comment]
@kismetgerald ah, you'll probably need to check out the source and build locally in that case. It's pretty straightforward, with instructions on the wiki.

**Comment by Kaian** [comment]
Heyo! 

The issue mentioned above by @davemidd that implements -R option will be ready for release 1.5.0. Most probably I will be releasing it tomorrow, as today I'm still on vacation hehe :angel:
As soon as release is published, I will schedule the job for making the package binaries, so stay tunned.

If you want to just test it, you can pull the latest master commit as @davemidd mentions.

Best regards!

**Comment by kismetgerald** [comment]
Thanks @davemidd 

but I'm a lazy Windows SysAdmin moonlighting in the *Nix world when nobody is watching.....ssshhhh.  So I think I'll wait for the package to be released.....lol

**Comment by Kaian** [comment]
Packages should be ready for 1.4.0

For the second request (integrate sngrep with a database engine) I think is a bit out of scope. We aim to make learning and debugging SIP a bit more easier and we highly recommend using each available tool for each task. Other projects (like HOMER as you mention) puts a lot of efforts in capturing, storing and displaying dialogs for later use in databases, create their proper indexes and so on.

Also, just for your information, for debian/ubuntu users there are _bleeding_ package repos that build each commit in case you want to test the latest version, just append _sngrep-bleeding_ to your distro name (p.ex: `deb http://packages.irontec.com/debian sngrep-bleeding-jessie main`), but be warned: those binaries are almost untested, everything may crash.

Thanks a lot for your time and suggestions!

**Comment by kismetgerald** [comment]
@Kaian Thanks for the kind response and suggestions.  I definitely understand why support db storage is out of scope.  I've used HOMER before and I liked what I saw so it'll probably be best, to stand up an instance and have a go at it.  I know FreeSWITCH supports packet capture directly to HOMER, so this should be fun.  I'm always eager to learn new things.

Anyway, thanks again.

By the way - did you mean packages are ready for v1.5.0 and not 1.4.0?
